### PR TITLE
[MISC] Replace stream operator by std::from_chars for Integral types.

### PR DESCRIPTION
**Pull request by smehringer**
long overdue and also has better handling for "out of range" errors.

### CODE REVIEW

File: include/seqan3/argument_parser/detail/format_parse.hpp [hunk 1]
```diff
@@ -344,25 +345,20 @@ class format_parse : public format_base
...
     void retrieve_value(option_t & value, std::string const & in)
     {
-        int64_t tmp;
-        std::istringstream stream{in};
-        stream >> tmp;
-
-        if (stream.fail() || !stream.eof()) // !stream.eof() catches 13a as wrong
-            throw type_conversion_failed("Argument " + in + " could not be casted to type " +
-                                         get_type_name_as_string(value) + ".");
+        auto res = std::from_chars(&in[0], &in[0] + in.size(), value);
```
Comment (eseiler) [inline include/seqan3/argument_parser/detail/format_parse.hpp]: Would `&in[in.size()]` also work?

File: include/seqan3/argument_parser/detail/format_parse.hpp [hunk 2]
```diff
@@ -344,25 +345,20 @@ class format_parse : public format_base
...
 
-        if (tmp > static_cast<int64_t>(std::numeric_limits<option_t>::max()) ||
-            tmp < static_cast<int64_t>(std::numeric_limits<option_t>::min()))
+        if (res.ec == std::errc::result_out_of_range)
             throw overflow_error_on_conversion("Argument " + in + " is not in integer range [" +
                                                std::to_string(std::numeric_limits<option_t>::min()) + "," +
                                                std::to_string(std::numeric_limits<option_t>::max()) + "].");
-
-        value = static_cast<option_t>(tmp);
+        else if (res.ec == std::errc::invalid_argument || res.ptr != &in[0] + in.size())
```
Comment (eseiler) [inline include/seqan3/argument_parser/detail/format_parse.hpp]: same here 🙈

File: include/seqan3/argument_parser/detail/format_parse.hpp [hunk 3]
```diff
@@ -344,25 +345,20 @@ class format_parse : public format_base
      */
     template <std::Integral option_t>
     //!\cond
-        requires IStream<std::istringstream, option_t>
+        requires IStream<std::istringstream, option_t> &&
+                 requires (option_t & v, const char * ptr) { std::from_chars(ptr, ptr, v); }
```
Comment (h-2) [inline include/seqan3/argument_parser/detail/format_parse.hpp]: You are already constraining option_t to be integral, do you need this additional check?

Comment (smehringer) [reply include/seqan3/argument_parser/detail/format_parse.hpp]: hmm if you define a type that models Integral it does not automatically work with from_chars does it? (But I actually noticed that this should reqire `arithmetic_concept` instead anyway.

Comment (h-2) [reply include/seqan3/argument_parser/detail/format_parse.hpp]: You cannot define a type that models std::Integral. std::Integral just covers the built-in types. Same for arithmetic

Comment (smehringer) [reply include/seqan3/argument_parser/detail/format_parse.hpp]: I removed the check.

**Pull request merged**